### PR TITLE
Use GitHub Actions for CI

### PR DESCRIPTION
**Pull request by ekohl**
This allows us to remove a workflow in Jenkins.

### CODE REVIEW

**Comment by ekohl** [comment]
Looks like Ruby 3.0 does some globbing that others don't. Mocking is such an awful tool.

**Comment by ekohl** [comment]
This time Ruby 3.1 failed. I suppose it's sort of random and suggests the test suite is already unreliable. The bigger matrix just uncovers it.

File: Gemfile [hunk 1]
```diff
@@ -3,8 +3,7 @@ source 'https://rubygems.org'
 gemspec
 
 group :development do
-  gem 'smart_proxy', git: 'https://github.com/theforeman/smart-proxy',
-                     branch: 'develop'
+  gem 'smart_proxy', github: 'theforeman/smart-proxy', branch: ENV['SMART_PROXY_BRANCH']
```
Comment (evgeni) [inline Gemfile:6]: In https://github.com/theforeman/smart_proxy_plugin_template/commit/71bd2b672e0ea2c848e2e96f14b121192411b2a9 we used `ENV.fetch(…, 'develop')` -- or will `branch` default to the default branch if it's `nil`?

Comment (ekohl) [reply Gemfile:6]: I tested it on Fedora without specifying it and that works. The Bundler docs haven't been updated, but looks like this is a Rubygem 2.3 feature: https://github.com/rubygems/rubygems/commit/6e032a9a6f1daf3ca08b1c17fe1770f51de54fca

Would you prefer to support < 2.3?

Comment (evgeni) [reply Gemfile:6]: I have no idea what < 2.3 means for rubygems 🙈
But also, we should be consistent between here and the docs in the actions repo and the template ;)

Comment (ekohl) [reply Gemfile:6]: * [2.7.0](https://www.ruby-lang.org/en/news/2019/12/25/ruby-2-7-0-released/) has 2.12
* [3.0.0](https://www.ruby-lang.org/en/news/2020/12/25/ruby-3-0-0-released/) has 2.2.3
* [3.1.0](https://www.ruby-lang.org/en/news/2021/12/25/ruby-3-1-0-released/) has 2.3.3

So I think we're not ready for that and I'll change it back to being explicit.

Comment (evgeni) [reply Gemfile:6]: Thanks for checking!

**Review by nofaralfasi — APPROVED** [review]

**Pull request merged**